### ISSUE TITLE
Feature request: Accept variables in the grid thresholds levels

### ISSUE THREAD

**Issue by toni-moreno** [grafana/grafana#922]
Hi @torkelo , I'm missing variable thresholds levels per dashboard template

Suppose I'm designing a product dashboard template (apache web server by example) , and I need watch when response time is greater than a threshold.

While the dashboard is the same , we can switch to other instances with variables , but response time ( if it is good or not )  it depends on the backend which real process the ( can be an application server, can be static html or perhaps a LAMP server) , in each case the  thresholds will differ a lot.

Also would be a good idea to map this  thresholds ( 2 digits)  with one of the variables in the template (instance name )  by example.

**Comment by Dieterbe** [comment]
not sure if this works at all, but maybe if we could create a linked $threshold variable that updates the threshold based on the variable setting, and then the threshold just uses $threshold

**Comment by biancalana** [comment]
+1 that will open a lot of new possibilities in terms of dynamic dashboards

**Comment by spijet** [comment]
I'd love to have an ability to map a metric to a threshold. I use Grafana to draw link usage graphs for network hardware and it's nice to have a threshold mapped to interface type. At the moment I use IfSpeed SNMP source to get active interface speed and draw it as a non-filled metric, but this way breaks dynamic max/min feature of Grafana.

**Comment by The-Nik** [comment]
+1

**Comment by alchuk** [comment]
Is there any news on this topic?

**Comment by ldormoy** [comment]
I also badly need this. Ideally the template variable's value could be the result of a query (not the path name but the value returned by this query)

**Comment by vaibhavinbayarea** [comment]
+1

Looking to set threshold value dynamically based on elasticsearch query.

**Comment by Corbyn** [comment]
For using Grafana as PNP4Nagios replacement it would be very helpful to have the possibillity to set warn/crit thresholds and min/max y-axis values based on field values!

**Comment by KawaiDesu** [comment]
Need this too.
For exmaple, i have memory graph and cant swtich $server variable. Servers have different amount of memory and i want to know, how much in GiB (not %) is used, but i don't remember total amount memory on them. So i need a threshold (or, maybe, Y-Max) with $total_mem variable.

**Comment by nanowish** [comment]
+1 here need accept $variable

**Comment by rasjoy** [comment]
+1

**Comment by kickkiwi** [comment]
I'm also eager for this feature, so that I can use a single graph template for multiple devices that have different thresholds.

**Comment by morsik** [comment]
I would like to have this feature too. What's current status of this?

**Comment by cattt84** [comment]
+1

**Comment by kmoczydlowski** [comment]
+1

**Comment by bogatuadrian** [comment]
+1

**Comment by sofixa** [comment]
+1

**Comment by golddiga** [comment]
+1

**Comment by timbru31** [comment]
@kmoczydlowski @bogatuadrian @sofixa @golddiga @cattt84 
Please stop bumping with +1, you are causing any subscriber a notification (and possibly an e-mail)

Use the reactions at the right of each comment to express your vote.
Thanks 😄

**Comment by gzll** [comment]
why can not use templating?

**Comment by RashmiSaravanan16** [comment]
Does dynamic threshold can be implemented to simpleJson data source .

**Comment by sanchomuzax** [comment]
+1

**Comment by adityasinha21** [comment]
+1

**Comment by kssminus** [comment]
+1

**Comment by gonzoj74** [comment]
+1

**Comment by anthraxn8b** [comment]
So now we have 2018. What about this feature?

**Comment by acalhounRH** [comment]
+1

**Comment by blah-dj** [comment]
+1 on this.

**Comment by B3DTech** [comment]
How has this been open for 4 years?  One of our use cases is in monitoring file storage quotas.  The threshold is defined as a soft limit and a hard limit, both of which are defined per file share, so setting this manually is impossible as the charts are dynamically created.  It seems trivial to let us chose a metric value to pass in as the value to the threshold.

**Comment by jfoxe** [comment]
Hi sillyfrog - Sorry I don't quite understand how I need to reference the new data source within the plugin (1.1.5) using elasticsearch queries (if these are supported).
I have tried adding in a new query, with a new alias and entering that into the threshold fields - obviously I am doing something incorrectly. 
The threshold fields within the plugin only allow integers and the letter 'e' in my instance.

**Comment by dsvensson** [comment]
I need this for checking if any job is lagging. As a workaround for now I use a template variable query max(my_prometheus_metric) and set that to a hidden variable, and then I have an extra query for my table view that does `abs(my_prometheus_metric - $variable)`, and add threhold on that column. It's not possible to hide a column that's used for setting row color unfortunately so I too would be happy for this feature.

**Comment by debmalyapan53** [comment]
Any update on this feature?

**Comment by sillyfrog** [comment]
@debmalyapan53 Sorry, I dropped the ball on this - but I have just put in another PR against the latest release. Hopefully it'll be merged in not too long.
Cheers.

**Comment by debmalyapan53** [comment]
Actually I want to implement a text box where user will be able to set threshold value, so that the same could be reflected in the graph.

**Comment by acaligiuri** [comment]
+1

**Comment by danielmotaleite** [comment]
using the template, we can workaround this, this is my setup:

![image](https://user-images.githubusercontent.com/11890049/57621239-70c61380-757a-11e9-8611-651ace630ec4.png)

![image](https://user-images.githubusercontent.com/11890049/57621423-e9c56b00-757a-11e9-9bbf-e865ee82bb0f.png)

Notice that i disabled one server in the last few minutes and the threshold changed

Sadly this do not solve everything, we can't "fill above" and can't have alerts based on this. So while the workaround works for many cases, it doesn't solve all the problems, grafana still needed to be able to use variables/queries to setup the thresholds/alerts

**Comment by ceinmart** [comment]
5 years and no news about this feat. request?

**Comment by smalenfant** [comment]
I'd like to alert on the Elastic ML results with Grafana. Example dashboard which I'd like to alert on. I have the lower/above threshold in the graph which are the model plot.

![image](https://user-images.githubusercontent.com/3803592/59234613-a3841b80-8bbb-11e9-9b49-48562cad2fb9.png)

**Comment by celster** [comment]
+1
this is a very important feature. Curios why is it lagging.

**Comment by polak785** [comment]
This is really a top required feature. Can we have an update on this please?

**Comment by arianofaoms** [comment]
> This is really a top required feature. Can we have an update on this please?

I second this. Please please please....

**Comment by ryanwaldron** [comment]
So basically this is never going to happen despite the fact that lots of people are asking for it here, and in a multitude of other requests, bug reports, and around the web in general. Way to take your users input and better your application.

**Comment by gcezaralmeida** [comment]
I need this too.

**Comment by ryantxu** [comment]
Check https://github.com/grafana/grafana/pull/19102  -- If you can use your variable to generate a value from a query, this could work.  "SELECT ${var} FROM xyz"

**Comment by finkr** [comment]
As a workaround, I do the following.  It assumes the second Y-axis is not used (or it has fixed min and max) value.

Exemple:
![image](https://user-images.githubusercontent.com/13090680/66703698-8603c180-ed15-11e9-8acc-46f2666c0777.png)

There's how to create it:
- Add a metric named "down" (or whatever) 
  - Use a query like this (multiple lines!). see notes below.
```
      1 - (day_of_week(vector(time())) > bool 0)
      * (day_of_week(vector(time())) < bool 6)
      * (((hour(vector(time()))  >= bool 7 - 2)))
      * (hour(vector(time())) < bool 19 - 2)
```
- Add a visualization override for this metric named "down":
  - Line fill: 1
  - Line width: 0
  - Staircase line: true
  - Color: _a light gray, like_ "rgb(167, 167, 167)"
  - Y-axis: 2
  - Z-index: 1 _(in front of other metrics)_
- Configure Y-Axis #2 visualisation:
  - min: 0
  - max: 1    (or whatever you want... multiply the metric above accordingly)

Note regarding the query:
* In the exemple query above, "7-2" is 7h00 in the morning, the "-2" is my time zone and daylight saving offset!
* You are welcome to reply to this comment if you adapt the query for other datasources

**Comment by louloizides** [comment]
Ugh... really need this and it's 2019... this is an old thread

**Comment by psy0rz** [comment]
same here..probably never going to happen?

**Comment by EggTronic** [comment]
Is it hard to implement or just no one want to do it? If so I would like to try

**Comment by liamdiprose** [comment]
@EggTronic I had a look for what it would involve, I think it comes down to the Threshold object:
https://github.com/grafana/grafana/blob/4a25ec0d36edcb551e6efd60e665df0fa93cb954/packages/grafana-data/src/types/threshold.ts#L1-L4

I was looking at implementing this for the bagguage specifically. The threshold(s) are used here:
https://github.com/grafana/grafana/blob/4a25ec0d36edcb551e6efd60e665df0fa93cb954/public/app/plugins/panel/bargauge/BarGaugePanel.tsx#L37

And further down in the same file, there's a `replaceVariables` function being used, I think its used to evaluate variables:
https://github.com/grafana/grafana/blob/4a25ec0d36edcb551e6efd60e665df0fa93cb954/public/app/plugins/panel/bargauge/BarGaugePanel.tsx#L53-L61

The way I can see it being done is
1. Modify the `Threshold` type so `value` can be a variable name (string) instead.
2. Change each panel type to call `replaceVariables` on any `Threshold.value` that are strings.
3. Make threshold setting box accept strings (variable names).

I'm not sure if it's the most appropriate design for the codebase though, Its a bit of a hack.

**Comment by CSimpiFoN** [comment]
+1 for the feature request

**Comment by rbrayner** [comment]
+1

**Comment by MaDOS** [comment]
I could really use this for the Gauge panel. I want to show a gauge for each taglogger writing into my database. Each logger reports its ingest rate and theoretical optimal ingest rate. But each logger needs different thresholds for their respective ingest rates.

**Comment by StaelensJarne** [comment]
+1

Is there a workaround for this kind of functionality (for the SingleStat panel)? Or can the thresholds be updated with a MySQL query?

**Comment by StaelensJarne** [comment]
What the status of these issue? Is this going to be implemented? Or is it to difficult to implement?

I have already asked this but whit no response: #25313

**Comment by gregormci** [comment]
+1

**Comment by luipir** [comment]
> I'm not sure if the design is appropriate for the codebase though, Its a bit of a hack.

@liamdiprose IMHO is appropriate, did you mind to prepose a pull request? 
A theoretical reason to avoid dynamic legends is that a panel template have to be referred to a uniform data set. Having dynamic legend open to have generic template that can manage different data stream and thsi could be not correct. IMHO I'm facing a use case where multiple indexes are sharing similar panel layout with different scales and color mapping.
Having dynamic thresholds allow me to manage a single or fre) dashboard instead of replicating if for different indexes.

Any feedback @torkelo or any other in grafana lab or core contributors ? some of the feature are managed in this PR https://github.com/grafana/grafana/pull/24763 ?

**Comment by liamdiprose** [comment]
@luipir Not yet, we need this feature badly at work but it's still on my backlog. :cry: 

Grafana 7 seems to be moving in the data-driven direction, so I believe this feature will be added eventually. @ryantxu of GrafanaLabs mentioned that using variables this way were upcoming in a similar issue-thread: https://github.com/grafana/grafana/issues/4929#issuecomment-601958742

I'd love to see every setting box accept variables. Maybe every user-supplied value should be "evaluated": Variable names evaluate to their value, Values evaluate to themselves. Variable evaluation will need the correct `replaceVariables` function (supplied to the panel as a prop) to work. 

It would be nice if `replaceVariables` was be replaced with `eval(varname) -> value`, and then `replaceVariables` can become a utility function that takes `eval` as a parameter.

**Comment by marian-aretu** [comment]
+1

**Comment by kyjanond** [comment]
+1

**Comment by scroodj** [comment]
+1

**Comment by scschneider** [comment]
+1

This would be an extremely useful feature when charting out stocks. For example, I can have thresholds based on a stock symbol's open price without charting it as a metric. The issue that I have ran into charting it as a metric is that it will flatten/normalize the graph.

Without charting open price:
![image](https://user-images.githubusercontent.com/25066231/98386497-18ec5980-2005-11eb-90e8-d78d52f2d832.png)

After charting open price:
![image](https://user-images.githubusercontent.com/25066231/98386694-5b159b00-2005-11eb-8078-1c96740c663b.png)

Setting the open price as a threshold (blue line):
![image](https://user-images.githubusercontent.com/25066231/98387164-edb63a00-2005-11eb-8db3-6ad7ac0d08dd.png)

**Comment by diepet** [comment]
+1

**Comment by ruudscheenen** [comment]
+1, this has been a long requested feature now and seems fairly 'standard'. Looking forward to a future update with this functionality!

**Comment by asrath** [comment]
+1

**Comment by etiennechabert** [comment]
Instead of the boring +1, I am putting a (giant) 🍺 as a bounty for the one bringing this feature to reality.

Usecase example: Trying to build a generic SLO dashboard for all the endpoints of an API, I would like the dashboard to receive SLO conditions via the URL of the dashboard.

**Comment by dmeehan1968** [comment]
I had a similar problem, I have a mysql query, and a panel with variables for min and max.  Not being able to use the min/max variables in the thresholds was disappointing.  But I managed to plot something similar, which at least allowed me to visualise the outliers.

1. In the query, reference your variables as additional columns.
  ```
  select
    time,
    data,
    ${minimum} as data_min,
    ${maximum} as data_max
  from table
  ```
2. Add a series override, with regex match on the min and max (e.g. `/data_(min|max)/`)
3. Set the override to:
  1. Enable lines (assuming the graph is otherwise points)
  2. Disable points
  3. Use a different colour (or use individual overrides for min/max)
  4. Disable legend
  5. Disable tool tips

If you have a metric column, you will probably want to name your data value as '' (`data as ''`) which leaves the legend/tooltip unadorned.

Of course this is just a visual workaround, you don't get the other benefits of thresholds, so this feature request is still much needed.

**Comment by ouinouin** [comment]
Hi, also need this functionality to be able to dynamically change the threshold in an industrial datalogging system
we should also accept basic calculations between variables in this field.

**Comment by luipir** [comment]
> Hi, also need this functionality to be able to dynamically change the threshold in an industrial datalogging system
> we should also accept basic calculations between variables in this field.

I abandoned grafana as OS proyect at least for his way to manage community... IMHO remain a good product but life is too short to struggle with marketing decisions to integrate community requests or PRs

**Comment by grutherford** [comment]
This would be REALY healpful for our org as we use variables for sales targets, thresholds (which we use to signify reaching target) needs to be manually updated currently when we change the variable.

**Comment by luipir** [comment]
Not sure if it's useful propose warkaround that give an excuse to avoid the implementation... My workaroud to have legend from DB data can be found in various panles. Just to point out an example with gauge panel where I get the legend from a custom JSON
https://github.com/luipir/GaliciaSustentable_deployment/blob/master/dashboards/luipir-gauge-panel/src/GaugePanel.tsx#L126

**Comment by a-vogel-tappert** [comment]
A threshold based on a query or variable would be great.
Please enter that feature :-)

**Comment by jason-turner2** [comment]
A work-around that's not-so-pleasing to the eye is to build it into the query, e.g.

```
SELECT (yourVar - highThreshold) FROM source etc etc etc.
```

And set the Grafana threshold to 0. Ugly and hard to interpret visually, but it gets the alerting working correctly. Having the chart paired with an easier-to-read chart makes it a bit less terrible.

**Comment by a-vogel-tappert** [comment]
I guess that workaround is not working if Flux language is used.

**Comment by jason-turner2** [comment]
I didn't give a Flux example because I don't use Flux much: I'm waiting until the dust settles, maybe around 2.1? But the docs say Flux can use multiple fields, and do math on them. I'd be surprised if it couldn't. Can those queries not be performed in Grafana?

**Comment by Alestrix** [comment]
I only need the visual, so @dmeehan1968 's solution to at least have a visual representation of a threshold (almost) solved my problem.
However, Influx complained that there must be at least one variable in each field, so my SELECT became this ugly hack

    SELECT $myvar + 0*first("somefield") as "my_threshold" FROM ...

to get around the error message.

**Comment by BerradaYassine** [comment]
Still needing this feature of variables in the threshold section, Any workaround for the time being?

**Comment by fboekema** [comment]
Is this already being worked on?

**Comment by JDA88** [comment]
This would be a very usefull feature for our organisation.
The visual workaround make us loose all the new threshold Time Series panel features!

**Comment by etiennechabert** [comment]
First things I want to say: **This situation is ridiculous**, since 2014 the community of Grafana is asking for this feature, the use case is OBVIOUS to anyone building dashboards seriously, this is the 3rd issue of the list when sorted by 👍... yet it's not prioritized or given any serious answer 🙈 

Now for the workaround I found (for CloudWatch, but should work with others), see screenshots below, long story short:
- Add a time-serie to your panel, with your variable as a value
- Define an override to change the rendering, so it looks like a threshold.

Here we are having a success-rate, put in comparison with his SLO given as a variable to the dashboard, see the query D on the bottom:
<img width="1449" alt="Screenshot 2021-10-01 at 10 37 57" src="https://user-images.githubusercontent.com/1692632/135591776-68978d74-d3bf-4375-9444-eb8c35cfb996.png">

And here is how I render it as a threshold
<img width="323" alt="Screenshot 2021-10-01 at 10 38 10" src="https://user-images.githubusercontent.com/1692632/135591949-a509d70d-4875-4816-9052-6e51082e383f.png">

**Comment by luipir** [comment]
@etiennechabert IMHO your solution is for dashboarding only, Im my case legend (e.g. colors depends on data range) so your solution do not cover the issue report, e.g. it is not a workaround. BTW thanks you for contributing to the discussion.... and I agree grafana community management is ridiculous!

**Comment by etiennechabert** [comment]
Agreed that it's a poor workaround... decided to share since it unblocked my case, hope it could help others, but yes in your case I have no idea how to go around the blocker :(

**Comment by alchuk** [comment]
@etiennechabert There's another downside of your workaround: natural thresholds do not affect panel scaling. Consider your meric much lower the configured threshold (e.g. 1-100Kbps on 100M network interface). A threshold line won't even be drawn until the metric gets closer to it, so you can see daily traffic fluctuation. Going your way you get 100M threshold line and another straight line on the bottom of the panel.

**Comment by pvinokurov** [comment]
+1

**Comment by Daniel-Mompean** [comment]
Hi Grafana team, any update please ? it would be very good !!!

**Comment by sandy724** [comment]
Can we have a timeline by when this feature would be available

**Comment by mjf** [comment]
Any progress with this **extremely needed** and **truly useful** feature, please?

**Comment by torkelo** [comment]
In Grafana 8.1 we launched two transformations that where designed to solve the underlying use case of dynamic thresholds: 

Config from query results: 
Docs: https://grafana.com/docs/grafana/latest/panels/reference-transformation-functions/#config-from-query-results
Demo: https://play.grafana.org/d/Fc6gwV4nz/transforms-config-from-query?orgId=1 

This allows you to use a normal panel query as a source of configuration for unit and thresholds that will be applied to another queries

The other transform: Rows to fields
Docs: https://grafana.com/docs/grafana/latest/panels/reference-transformation-functions/#rows-to-fields
Demo: https://play.grafana.org/d/T5dVQV47z/transforms-rows-to-fields?orgId=1 

This transform is a bit different, here you can form a single query result use one column as a source of config for another column. Say you have a Value, Threshold, Unit columns you can set the thresholds and unit for the value field using the values from the threshold and unit columns. The result from the transform is a new table structure where each row in the source table becomes a new field in the result table.

**Comment by simPod** [comment]
@luipir bro, you can go and contribute. it's oss.

**Comment by torkelo** [comment]
> here where serious architecture limitation?
@luipir well sort of. Introducing the use of variables in thresholds felt like the wrong tool for the job. Variables main purpose is filtering data, not for setting display settings. Making it possible to set variables inside thresholds would be very messy from a platform perspective (change it's type to string, constantly have to interpolate it before evaluations, impact validation logic etc). 

And while digging into the biggest underlying request was a need to set thresholds dynamically from the result of a query. And the new transform system we introduce in v7 does support setting data config (thresholds etc) as well as those are attached to the data structure. This is why we solved this by introducing new transforms that allow you to query for thresholds, or use other fields in the same result set as source for a threshold

**Comment by luipir** [comment]
@simPod I'm used to contribute to OSS project but depends a lot of their community politicis. If you read the thread, you can see that I already gave my little piece of contribution..

**Comment by Daniel-Mompean** [comment]
@torkelo, thanks for any feedback.

I would like to share a need of mine, today we have a process performance indicator per hour and per user, and we should show this by colors.

however, we have clients in more than 160 countries, and each operation has its performance level, the KPI is one for the entire system/world, each user selects his warehouse in a variable.

therefore, if we had the possibility to make the threshold flexible, it would be a giant advance.

please help us to keep Grafana as the standard Dashboard tool in our company, we are present in more than 160 countries, and we have already developed Dashboards to monitor performance in several areas.

![kpi](https://user-images.githubusercontent.com/68402165/156759049-32563643-cfb9-4bd5-95e5-ffe6a376628a.PNG)

**Comment by torkelo** [comment]
Did you read how the new transforms work? they allow you to query for thresholds

**Comment by walkeran** [comment]
@torkelo Thank you very much for the work you've done on transformations! We've been using them for only a short while, but they've solved this specific issue for us, as well as a couple of other things we were wanting to accomplish. Furthermore, they appear to be powerful enough that I can imagine they'll be a quite handy solution for problems that we haven't even come across yet.

Hats-off to Grafana, and thank you to you and your team for continuing to create and maintain such a great piece of software!

**Comment by Daniel-Mompean** [comment]
tks, does it work with postgres?

**Comment by Daniel-Mompean** [comment]
I got it now, worked, thank you a lot.

**Comment by shehrozeee** [comment]
> Did you read how the new transforms work? they allow you to query for thresholds

Thank you. I read it and I was able to get dynamic thresholds. and there is a lot that can be done using transformations.

**Comment by JDA88** [comment]
I'm playing with `Config from query results`, it works but i can only set up `Threshold1` is there a way to have more Thresholds steps?

**Comment by shehrozeee** [comment]
For multiple thresholds I just mapped another value to threshold1, it works
for making the lines at least.

**Comment by JDA88** [comment]
Are you using PromQL ? Do you mind sharing a panel example? I already tried setting multiple query values to `Threshold1` but only the first one seen to be taken into account.

**Comment by JDA88** [comment]
Ok I have played a little with the objective to use dashboard variables as threshold with Prometheus queries.
Context:

- A dashboard with two variable that can be used as threshold (`$threshold1`, `$threshold2`) on a Time series panel. 
- _Should I open a different issue? because this one is related to grid_

`--------------------------------------------------------------------------------------------`
First scenario (using “**Use as Threshold1**”):

- PomQl (Format Table & Instant):
```
label_replace(
 label_replace(
 (MIN by (instance) (abs (up{instance="$instance"}-1)))
 , "Level1", "$threshold1", "", "")
, "Level2", "$threshold2", "", "")
```
Transform:
- Apply to: Fields with type
- Apply to options: Numeric
- Level1 | Threshold1 | Last
- Level2 | Threshold1 | Last

Result:
- Threshold are correctly displayed and are dynamic
- All Threshold levels are red, I didn’t found a way to change the colors

`--------------------------------------------------------------------------------------------`
Second scenario (using “**Use as Threshold value mapping**”):

- PomQl (Format Table):
```
label_replace(
 label_replace(
 (MIN by (instance) (abs (up{instance="$instance"}-1)))
 , "Value", "$threshold1", "", "")
, "Color", "orange", "", "")
or
label_replace(
 label_replace(
 (MIN by (instance) (abs (up{instance="$instance"}-1)))
 , "Value", "$threshold2", "", "")
, "Color", "red", "", "")
```
Transform:

- Color | Value mappings / Color | All values
- Values | Value mappings / Values | All values

Result:

- Threshold are not displayed, i'm probably missing something

**Comment by zoarax** [comment]
+1

**Comment by sanchomuzax** [comment]
> Did you read how the new transforms work? they allow you to query for thresholds

I’ve tried to understand how this works, but I can’t find a step-by-step example from which I understand how to use it. Can you help me in this? Thank you!

**Comment by torkelo** [comment]
Closing this, see this comment for how to have dynamic threshold 
https://github.com/grafana/grafana/issues/922#issuecomment-1058424892

**Comment by samjewell** [comment]
@torkelo thanks so much for showing how this feature works in your comment above.

> Config from query results:
Docs: https://grafana.com/docs/grafana/latest/panels/reference-transformation-functions/#config-from-query-results
Demo: https://play.grafana.org/d/Fc6gwV4nz/transforms-config-from-query?orgId=1

The first demo doesn't seem to be working right now - the data seems to be missing:

![image](https://user-images.githubusercontent.com/2903904/183449936-ed191fbb-4d47-450e-94a0-954638dee7df.png)

I just thought I'd ask in case this is easy to fix (or implement on some data that still exists on play.grafana.org). Thanks again!

**Comment by gadgadskiy** [comment]
> In Grafana 8.1 we launched two transformations that where designed to solve the underlying use case of dynamic thresholds:
> 
> . . . . . . . .

Hi, @torkelo ! Currently, these links lead to a demo for Grafana v9. I have Grafana v8 and can't see options like "Scenario: CSV Content," only SQL queries.
Is it possible to do something similar in Grafana 8? (I'm interested in "Min, max, threshold from separate query": https://play.grafana.org/d/Fc6gwV4nz/transforms-config-from-query?orgId=1&editPanel=2)

**Comment by jon-ton** [comment]
> Config from query results:
> Docs: https://grafana.com/docs/grafana/latest/panels/reference-transformation-functions/#config-from-query-results
> Demo: https://play.grafana.org/d/Fc6gwV4nz/transforms-config-from-query?orgId=1

Checked out the Docs above, but whenever I use Threshold1 to make the value dynamic, I lose my color preferences. For example, in my use case I need red on the bottom and green on top, however once I use the config transformation it overwrites that color preference. 

I couldn't get any of the Threshold overrides to change just the color, without overwriting the value as well. 

Appreciate all the updates on transformations though!

**Comment by baur** [comment]
Same problem 
 I need red on the bottom and green on top

![image](https://github.com/grafana/grafana/assets/2898361/4ba8d91f-822f-4731-98e6-fe14e2f69aea)

**Comment by yurividal** [comment]
> Same problem: I need red on the bottom and green on top
> 
> ![image](https://user-images.githubusercontent.com/2898361/239837566-4ba8d91f-822f-4731-98e6-fe14e2f69aea.png)

Same here. My threshold is actually a minimum ammount, so, i would like my panel to go RED if it goes bellow the threshold and green (or blue) if it goes above. 

Why not just let us set a variable into the threshhold field?!

**Comment by mpierrau** [comment]
Any news on the issue with colors?

**Comment by iambenmillane** [comment]
The ability to set a base colour of your choosing is definitely something that'd be good to address. It defaults to green, which is actually against the best practice video for dashboard colors that Grafana put out (suggesting blue and red due to colour blind accessibility.)

I've not been able to figure out how to set multiple thresholds or the base colour dynamically yet.

**Comment by daveshm** [comment]
I also would love this feature for dashboarding SLO info as a variable. I would use multiple threshold and base color configuration where I have a green/yellow/red visualization with the SLO percent value being red and (1 - slo_percent) / 2 + slo_percent being the yellow. @yurividal's comment `Why not just let us set a variable into the threshhold field?!` is spot on
